src/main/java/entities/CollaborativeTask.java

```diff
@@ -0,0 +1,217 @@
+package Entities;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+public class CollaborativeTask extends Task implements Timeblockable{
+    private boolean recurring;
+    private String frequency;
+    private LocalDateTime startTime;
+    private LocalDateTime endTime;
+    private ArrayList<StudentUser> teammates;
+    private ArrayList<StudentUser> pendingTeammates;
+    private ArrayList<StudentUser> declinedTeammates;
+    private StudentUser leader;
+
+
+    /**
+     * Create a CollaborativeTask with a title
+     * If the collaborative task is recurring, indicate the frequency (eg "daily", "weekly", "monthly)
+     * Otherwise, the frequency is blank
+     * @param title - the title of the Collaborative Task
+     * @param id - the unique ID of the Collaborative Task
+     * @param recurring - whether the Collaborative Task is recurring
+     * @param frequency - the frequency at which the Collaborative Task occurs (if recurring)
+     * @param creator - the Student User who creates the Collaborative Task
+     */
+    public CollaborativeTask(String title, String id, boolean recurring, String frequency, StudentUser creator) {
+        super(title, id);
+        this.recurring = recurring;
+        if (recurring) this.frequency = frequency;
+        else this.frequency = "";
+        this.leader = creator;
+    }
+
+    /**
+     * Create a Collaborative Task with a title and priority
+     * If the collaborative task is recurring, indicate the frequency (eg "daily", "weekly", "monthly)
+     * Otherwise, the frequency is blank
+     * @param title - the title of the Collaborative Task
+     * @param id - the unique ID of the Collaborative Task
+     * @param priority - the priority value of the Collaborative Task
+     * @param recurring - whether the Collaborative Task is recurring
+     * @param frequency - the frequency at which the Collaborative Task occurs (if recurring)
+     * @param creator - the Student User who creates the Collaborative Task
+     */
+    public CollaborativeTask(String title, String id, int priority, boolean recurring, String frequency, StudentUser creator) {
+        super(title, id, priority);
+        this.recurring = recurring;
+        this.frequency = frequency;
+        this.leader = creator;
+    }
+
+    /**
+     * @return leader of the Collaborative Task
+     */
```

Comment:
For the scope of methods: in Task I originally made all of my methods protected, but realized when it came time to work on my use cases that they should have been made public. 

Unless it breaks some major clean architecture rules, I think its safe to make your methods public too.

Reply:
Yeah, I noticed that too